src/rust/engine/options/src/args.rs

```diff
@@ -143,25 +145,74 @@ impl Args {
     }
 }
 
+pub(crate) struct ArgsTracker {
+    unconsumed_args: Mutex<HashSet<Arg>>,
+}
+
+impl ArgsTracker {
+    fn new(args: &Args) -> Self {
+        Self {
+            unconsumed_args: Mutex::new(args.args.clone().into_iter().collect()),
```

Comment:
We could have used `&'a str` references into the `Args`'s internals, with 'a being the lifetime of the `Args` instance, but that seems like overkill to avoid a tiny bit of string copying. Command lines are usually very short, and even unusually long ones would only spend microseconds on these copies.